freezegun/api.py

```diff
@@ -35,6 +38,60 @@
     import copyreg
 
 
+# keep a cache of module attributes otherwise freezegun will need to analyze too many modules all the time
+_GLOBAL_MODULES_CACHE = None
+
+
+def _get_global_modules_cache():
+    global _GLOBAL_MODULES_CACHE
+    if _GLOBAL_MODULES_CACHE is None:
+        _GLOBAL_MODULES_CACHE = _setup_modules_cache()
+    return _GLOBAL_MODULES_CACHE
+
+
+def _get_module_attributes(module):
+    result = []
+    for attribute_name in dir(module):
+        try:
+            attribute_value = getattr(module, attribute_name)
+        except (ImportError, AttributeError, TypeError):
+            # For certain libraries, this can result in ImportError(_winreg) or AttributeError (celery)
+            continue
+        else:
+            result.append((attribute_name, attribute_value))
+    return result
+
+
+def _setup_modules_cache():
+    # FIXME: move this definition to be at the top-level
+    real_time_object_ids = set(id(obj) for obj in real_date_objects)
+    result = {}
+    for mod_name, module in list(sys.modules.items()):
+        # ignore this module
+        if mod_name == __name__:
+            continue
+        date_attrs = []
+        all_module_attributes = _get_module_attributes(module)
+        for attribute_name, attribute_value in all_module_attributes:
+            if id(attribute_value) in real_time_object_ids:
+                date_attrs.append((attribute_name, attribute_value))
+        result[mod_name] = (_get_module_attributes_hash(module), date_attrs)
+    return result
+
+
+def _get_module_attributes_hash(module):
+    return '{0}-{1}'.format(id(module), hashlib.md5(','.join(dir(module)).encode('utf-8')).hexdigest())
+
+
+def _get_cached_module_attributes(mod_name, module):
+    global_modules_cache = _get_global_modules_cache()
+    module_hash, cached_attrs = global_modules_cache.get(mod_name, ('0', []))
+    if _get_module_attributes_hash(module) == module_hash:
+        return cached_attrs
+    else:
+        return _get_module_attributes(module)
```

Comment:
Do we want to be setting these back on `global_modules_cache` to update the cache?